implant/sliver/ps/ps_linux.go

```diff
@@ -89,6 +94,34 @@ func getProcessCmdLine(pid int) ([]string, error) {
 	return argv, nil
 }
 
+func getProcessArchitecture(pid int) (string, error) {
+	exePath := fmt.Sprintf("/proc/%d/exe", pid)
+
+	f, err := os.Open(exePath)
+	if err != nil {
+		return "", err
+	}
+	_, err = f.Seek(0x12, 0)
+	if err != nil {
+		return "", err
+	}
+	mach := make([]byte, 2)
+	_, err = io.ReadAtLeast(f, mach, 2)
```

Comment:
We should probably not discard `n` returned by `io.ReadAtLeast` and make sure it's 2, as expected. We should also check `err` before the return statement, otherwise we might have a nil pointer deref in the following `if` statements.